### DISCUSSION TITLE
can not create client, bad configuration when using terraform apply command

### DISCUSSION THREAD

**Discussion by kaliker** [chadgeary/cloudblock#80] [Q&A] [answered]
after running the command:
oci iam compartment list --all --compartment-id-in-subtree true --access-level ACCESSIBLE --include-root --raw-output --query "data[?contains(\"id\",'tenancy')].id | [0]"

I'm getting a value so the config file is set properly.
when using terraform apply command I keep getting the error: 
"can not create client, bad configuration: did not find a proper configuration for private key"

what am I doing wrong?

Thanks

**Comment by kaliker** [comment, marked as answer]
The problem was with the private key password. 
I removed the password from the private key and then I was able to continue without any problem.

**Reply by NSidious** [reply]
Hi Kaliker... I'm getting the same error. How do I remove the password from the private key?

**Reply by neitsab** [reply]
https://letmegooglethat.com/?q=remove+password+from+private+key 😁

tldr:

    ssh-keygen -p
